client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx

```diff
@@ -65,6 +71,9 @@ export const AppsSubmissionForm: React.FC = () => {
   const {
     user: { username },
   } = useAuthenticatedUser() as { user: TUser };
+  const { data: jobData } = useGetJob('select', { uuid: jobUUID }) as {
```

Comment:
There's a slight delay in populating the app form while the job data is being fetched. Would it be possible to keep the loading state while this is being requested as well? Might need to be a suspense query?

Reply:
good point, will look into switching to suspense.

Reply:
done, moved to suspense